wellcomeml/ml/frequency_vectorizer.py

```diff
@@ -115,16 +115,9 @@ def transform(self, X, regex=True, spacy_lemmatizer=True, *_):
 
         return super().transform(X)
 
-    def fit(self, X, regex=True, spacy_lemmatizer=True, *_):
-        if regex:
-            X = self.regex_transform(X)
-        if spacy_lemmatizer:
-            X = self.spacy_lemmatizer(X)
-
-        logger.info("Fitting vectorizer.")
-
-        X = [" ".join(text) for text in X]
-
-        super().fit(X)
-
+    def fit(self, X, *_):
+        super().fit()
         return self
+
+    def fit_transform(self, X, y=None):
```

Comment:
isn't this provided for free from sklearn?

Reply:
That's why the `TfId` class in sklearn has to implement their own `.fit.` even though it's a transform mixin.

Reply:
👍